File: Sorting/03_Insertion_Sort/python/insertionSort.py
def insertionSort(lst):
    length_of_list = len(lst)
    for i in range(1, length_of_list):
        value = lst[i]
        hole = i
        while hole > 0 and lst[hole - 1] > value:
            lst[hole] = lst[hole - 1]
            hole = hole - 1
        lst[hole] = value
    return


if __name__ == '__main__':
    unsorted_list = [2, 7, 4, 1, 5, 3]
    print('unsorted list is: {}'.format(unsorted_list))
    insertionSort(unsorted_list)
    print('sorted list is: {}'.format(unsorted_list))

File: Sorting/03_Insertion_Sort/cpp/InsertionSort.cpp
#include<iostream>
using namespace std;

void insertionSort(int arr[], int n) {
	for (int i = 1; i < n; i++) {
		int value = arr[i];
		int hole = i;
		while (hole > 0 && arr[hole - 1] > value) {
			arr[hole] = arr[hole - 1];
			hole -= 1;
		}
		arr[hole] = value;
	}
}

int main() {
	int arr[] = { 2, 7, 4, 1, 5, 3 };
	int sizeOfArray = sizeof(arr) / sizeof(int);

	cout << "unsorted array: ";
	for (int i = 0; i < sizeOfArray; i++) {
		cout << arr[i] << " ";
	}

	insertionSort(arr, sizeOfArray);

	cout << "\nsorted array: ";
	for (int i = 0; i < sizeOfArray; i++) {
		cout << arr[i] << " ";
	}
}
